mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxViewModelProvider.kt

```diff
@@ -28,10 +30,26 @@ object MvRxViewModelProvider {
         stateClass: Class<out S>,
         viewModelContext: ViewModelContext,
         key: String = viewModelClass.name,
-        initialStateFactory: MvRxStateFactory<VM, S> = RealMvRxStateFactory()
+        initialStateFactory: MvRxStateFactory<VM, S> = RealMvRxStateFactory(),
+        viewModelFactory: MvRxFactory<VM> = MvRxFactory { createViewModel(key, viewModelClass, stateClass, viewModelContext, initialStateFactory = initialStateFactory) }
     ): VM {
+        // check if this needs to be recreated from saved state due to process being restarted
+        val restoreViewModelFactory = viewModelContext.savedStateRegistry.consumeRestoredStateForKey(key)?.let { savedStateBundle ->
+            val restoredArgs = savedStateBundle.get(KEY_MVRX_SAVED_ARGS)
+            val restoredState = savedStateBundle.getBundle(KEY_MVRX_SAVED_INSTANCE_STATE)
+
+            requireNotNull(restoredState) { "State was not saved prior to restoring!"}
+
+            val restoredContext = when (viewModelContext) {
+                is ActivityViewModelContext -> ActivityViewModelContext(viewModelContext.activity, restoredArgs)
+                is FragmentViewModelContext -> FragmentViewModelContext(viewModelContext.activity, restoredArgs, viewModelContext.fragment)
+            }
+            MvRxFactory { createViewModel(key, viewModelClass, stateClass, restoredContext, restoredState::restorePersistedState) }
```

Comment:
this pattern `MvRxFactory { createViewModel` seems to be used several times, there could be a function or constructor to simplify it to just do `createMvRxFactory(key, viewModelClass, stateClass...)`